--- linux/usr/gen_initramfs_list.sh.orig	2017-11-10 01:50:25.685322259 -0500
+++ linux/usr/gen_initramfs_list.sh	2017-11-10 02:24:42.707363462 -0500
@@ -192,7 +192,21 @@
 # if a regular file is specified assume it is in gen_initramfs format
 input_file() {
 	source="$1"
-	if [ -f "$1" ]; then
+	if [ -x "$1" ]; then
+		${dep_list}header "$1"
+		if [ -z ${dep_list} ]; then
+			print_mtime "$1" >> ${output}
+			"$1" >> ${output}
+		else
+			echo ".config \\"
+			echo "$1 \\"
+			"$1" | while read type dir file perm ; do
+				if [ "$type" = "file" ]; then
+					echo "$file \\";
+				fi
+			done
+		fi
+	elif [ -f "$1" ]; then
 		${dep_list}header "$1"
 		is_cpio="$(echo "$1" | sed 's/^.*\.cpio\(\..*\)\?/cpio/')"
 		if [ $2 -eq 0 -a ${is_cpio} = "cpio" ]; then
